packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch

```diff
@@ -0,0 +1,50 @@
+From dedfae673e28a1605419ec6a6aab67a5c9f039ae Mon Sep 17 00:00:00 2001
+From: Sparks Song <shijiao@amazon.com>
+Date: Wed, 4 Dec 2024 23:47:28 +0000
+Subject: [PATCH] mkfs: source defaults from config file to make nrext64
+ default off on multiple kernel versions
+
+---
+ mkfs/xfs_mkfs.c | 19 ++++++++++++++++++-
+ 1 file changed, 18 insertions(+), 1 deletion(-)
+
+diff --git a/mkfs/xfs_mkfs.c b/mkfs/xfs_mkfs.c
+index 6d2469c..9823571 100644
+--- a/mkfs/xfs_mkfs.c
++++ b/mkfs/xfs_mkfs.c
+@@ -39,6 +39,9 @@
+  */
+ #define WHACK_SIZE (128 * 1024)
+
++/* Default path for the mkfs.xfs configuration file */
++#define DEFAULT_CONFIG_PATH "/usr/share/xfs/mkfs.xfs.conf"
++
+ /*
+  * XXX: The configured block and sector sizes are defined as global variables so
+  * that they don't need to be passed to getnum/cvtnum().
+@@ -4428,7 +4431,21 @@ main(
+ 	 * the options from this file parsed, we can then proceed with parameter
+ 	 * and bounds checking and making the filesystem.
+ 	 */
+-	cfgfile_parse(&cli);
++
++	if (cli.cfgfile == NULL) {
++		/* No user config file specified, try default config */
++		FILE *default_config_file = fopen(DEFAULT_CONFIG_PATH, "r");
++		if (default_config_file != NULL) {
++			fclose(default_config_file);
```

Comment:
This feels a bit odd to `fopen()` the file only for checking if it exists, then close it, and finally parse it. Would it make more sense to implement similar logic to `cfgfile_parse()` but not error out completely if the file isn't present/parsable?